src/components/ThreejsEditorModal.jsx

```diff
@@ -146,37 +159,130 @@ export class ThreejsEditorModal extends ModalDialog {
         this.editor.signals.objectSelected.dispatch(this.editor.camera);
     }
 
-    showAlert() {
-        swal({
-            icon: "error",
-            buttons: {
-                cancel: "Cancel",
-                exit: "Exit",
-            },
-            text: "Unable to extract a material from the editor!",
-        }).then((value) => {
-            switch (value) {
-                case "exit":
-                    super.onHide();
-                    break;
-                default:
-                    break;
-            }
+    /**
+     *  shows alert with specific parameters
+     *  @typedef {{ text: string, onClick: Function }} ButtonsType
+     *  @typedef {{ title: string, content: string, buttons: ButtonsType }} ShowAlertInputType
+     *  @param {ShowAlertInputType} params
+     */
+    showAlert(params) {
+        this.alertRef.open(params);
+    }
+
+    /**
+     * submits multiple selections group
+     */
+    submitMultipleSelectionGroup() {
+        this.editor.execute(
+            new SubmitMultipleSelectionCommand(this.editor, this.editor.multipleSelection),
+        );
+    }
+
+    /**
+     * removes multiple selection group
+     */
+    removeMultipleSelectionGroup() {
+        this.editor.execute(new RemoveMultipleSelectionGroupCommand(this.editor));
+    }
+
+    /**
+     * this function shows confirm window if user forgets to submit multiple selection and tries to exit from editor
+     */
+    showSubmissionMultipleSelectionModal() {
```

Comment:
If multiple selection is handled inside the 3DEditor/wave.js, should we move this logic there as well? Otherwise we seem to be managing the state of that component from outside of it.

Reply:
Don't think that is really significant, I guess we can use both ways, why I implemented it in that way because the wave already has a react and it's easier to manipulate with DOM, and also we have logic that extracts data from 3DEditor outside.

Reply:
Let's add a "TODO"